### ISSUE TITLE
Пофиксить в коде гарантирование долга по золотому

### ISSUE THREAD

**Issue by t3ran13** [GolosChain/golos#817]
согласно посту
https://golos.io/ru--golos/@litrbooh/ru-pochemu-delegaty-podryvayut-ideyu-stabilxnogo-aktiva-ili-pochemu-nuzhno-ubratx-proczenty-po-gbg

> Но есть и второе ограничение, если масса SBD/GBG превышает 10% от капитализации golos/steem, то актив уже не будет обеспечиваться номиналом. Но если курс катастрофически не падает, на стиме такого эффекта трудно добиться, так как эмиссия остановится на 5%.

золотой должен быть обеспечен заявленым номиналом при любом долге системы как заявлено в белой бумаге
https://wiki.golos.io/4-documentation/golos_whitepaper.html#%D0%B7%D0%BE%D0%BB%D0%BE%D1%82%D0%BE%D0%B9-%D0%BE%D0%B1%D0%B5%D1%81%D0%BF%D0%B5%D1%87%D0%B5%D0%BD%D0%BD%D1%8B%D0%B9-%D0%B3%D0%BE%D0%BB%D0%BE%D1%81%D0%B0%D0%BC%D0%B8

**Comment by bitfag** [comment]
Это консенсусная вещь, прежде чем трогать код надо договориться чего мы хотим. Касательно вопроса, поддерживаю идею, можно для начала задрать максимальный долг системы например с 10 до 20%.

**Comment by litrbooh** [comment]
@bitfag 
>  Касательно вопроса, поддерживаю идею, можно для начала задрать максимальный долг системы например с 10 до 20%.
А смысл лимитов тогда? :)

**Comment by t3ran13** [comment]
Я бы лимиты убрпл вообще, если люди активничают, то пускай долд продолжает расти.
Но это немного другая история, сейчас вопрос поыиксить нижнюю планку

**Comment by t3ran13** [comment]
Rimus, [28.08.18 19:26]
Господа делегаты - прошу помочь найти ошибку в расчёте цены прайсфида GBG/GOLOS.

Ну т.е. это я что-то не понимаю или блокчейн всё же не корректно считает median_sbd_price.

Суть вопроса проста: уже неделю как у всех делегатов из топ-21 курс плюс-минус 0.8 GBG за 1 GOLOS, в то время как в блокчейне

  "median_sbd_price": {
    "base": "162361263.213 GBG",
    "quote": "144202190.926 GOLOS"
  }

т.е. 1 GBG за 0.88 GOLOS что в 1.42 раза выше, чем поставляют делегаты. Можете проверить по своим фидам.

У меня глюки или блокчейн как-то не корректно прайс фид считает?

vvk, [28.08.18 19:28]
[In reply to Rimus]
Уже очень давно работает ограничение на 10% долга, оно не даёт снизить цену ниже лимита

vvk, [28.08.18 19:28]
% ./estimate_gbg_debt.py
INFO: Minimal possible median price GBG/GOLOS: 1.126
INFO: System GBG debt percent: 10.000

T3RAN13, [28.08.18 19:35]
[In reply to vvk]
чет я не понимаю, почему долг влияет на прайсфид?

T3RAN13, [28.08.18 19:37]
я понимаю когда ограничивается выпуск гбг, но причем тут прайсфид? голоса нельзя конвертить в золотые, потому какой смысл в ограничениях прайсфида??

vvk, [28.08.18 19:42]
[In reply to Rimus]
Ну это ещё Лаример придумал, что долговые обязательства системы обеспечены только 10% капитализации основного ассета

T3RAN13, [28.08.18 19:43]
[In reply to vvk]
бредовая логика. т.е. много гбг плохо, но уменьшить мы их вам не дадим? т.е. с двух сторон подпорка

T3RAN13, [28.08.18 19:43]
[In reply to vvk]
это явно нужно фиксить

T3RAN13, [28.08.18 19:45]
она тупо создает платину при конвертации гбг в голоса. абсурд!

T3RAN13, [28.08.18 19:46]
ввиду таких больших стаканов на бирже, проще конвертировать и пустить в силу. но вот это митнимальное условие , кривое, не дает этого сделать

T3RAN13, [28.08.18 19:48]
никакого лимита быть не должно

T3RAN13, [28.08.18 19:48]
можно просто сделать % эммисии за период времени чб не превышал сколько то

T3RAN13, [28.08.18 19:49]
чб за 3 дня не выдать годовую эммисию

T3RAN13, [28.08.18 19:49]
но только в плате за посты

T3RAN13, [28.08.18 19:49]
аконвертить то логично разрешить в голоса

vvk, [28.08.18 19:51]
Давно уже ниже номинала вообще-то, конвертить не выгодно

vvk, [28.08.18 19:51]
INFO: External price BTC/gold: 0.00000550
INFO: External price BTC/GBG: 0.00000462
INFO: GBG/gold rate: 0.84

**Comment by primusd** [comment]
Ссылки на участки кода не помешают для полноценного анализа.

**Comment by afalaleev** [comment]
ссылка на участок кода:
https://github.com/GolosChain/golos/blob/a3913e9988beabebc1a72dace269474df7d1ef4c/libraries/chain/database.cpp#L3612-L3620

**Comment by bitfag** [comment]
@t3ran13 резюмируя - ты предлагаешь полностью убрать ограничение по min_price и в любом случае конвертировать GBG эквивалентно цене 1мг ?

**Comment by primusd** [comment]
> @t3ran13 резюмируя - ты предлагаешь полностью убрать ограничение по min_price и в любом случае конвертировать GBG эквивалентно цене 1мг ?

@t3ran13 наверняка сам прокомментирует, но с моей точки зрения именно такой подход (убрать ограничение по min_price) в любом случае стоит серьёзно рассмотреть.

Как минимум по двум причинам:

1) В текущем Голосе это активирует GBG маркет и вообще, всё, что связано с GBG. Что в целом положительно для блокчейна, когда у него используются оба актива (GOLOS и GBG) изначально описанные в WP и коде. Тем более, что по GBG есть крупная биржа с листингом.

2) С учётом перспективных планов по отказу от GBG в случае перехода на будущий GEOS фиксированная планка в min_price может вызвать кучу критики когда придёт время собственно перевода GBG в GOLOS - однозначно будут претензии от сообщества и потенциальных инвесторов в связи с некорректным курсом GBG/GOLOS. Пусть это будет регулироваться самим рынком - тогда претензий не возникнет.

**Comment by t3ran13** [comment]
> @t3ran13 резюмируя - ты предлагаешь полностью убрать ограничение по min_price и в любом случае конвертировать GBG эквивалентно цене 1мг ?

да, именно. @primusd правильно пишет, дополню 
- **У нас гарантия в белой бумаге записана, а мы ее не выполняем. **
- я отправляю одни фиды в бч, как делегат, а работают другие. ппц какоей-то.


**Давайте организуем отдельный дискорд по этому вопросу и вопросу долга системы тоже?**
Мы сейчас в ловушке, у нас затор золотых в экономике, они не конвертируются изса катсыля с min_price и не штампуются дальше из-за ограничения долго системы.

P/S
 Я бы эти два кастыля убрал нафиг

**Comment by bitfag** [comment]
@t3ran13 согласен, собрать дискорд и инициировать голосование делегатов по внесению изменений в логику.

**Comment by afalaleev** [comment]
Я бы все таки не называл это багом. 
В белой бумаге не оговорено наличие границы, обрабатывающей ситуацию черного лебедя.
Но там точно также не оговоренно наличие процентной ставки на GBG, которая и стала причиной достижения ситуации лебедя.
Белая бумага описывает намерения, а не полностью оговаривает параметры экономики - иначе ее нельзя было бы вообще никакие параметры экономики менять.

Изначальная реализация в Стиме при достижении ситуации лебедя уменьшала количество GBG на счетах и в операциях конвертации, но из за возможных проблем с округлением, было переложено на плечи делегатов - через уменьшение процента роста.

Можно не снимать нижнюю границу в 10%, можно останавливать рост по процентной ставке.
Но это как решит сообщество.

**Comment by litrbooh** [comment]
@afalaleev @afalaleev 
Это или бага, или скам. Выбирайте сами.

Вопрос вот в чём:
1) У вас введен лимит на 10% от капитализации, при достижении GBG перестают выплачивать авторам. Но держателям GBG платить продолжают. Да, можно свалить это на делегатов, но если делегаты "странноватые", то система должна выключать ЛЮБУЮ эмиссию GBG при достижении 10% долга.

2) И уже жестко ограничив эмиссию на 10% (без учета курсовых рисков), можно четко зафиксировать стоимость золотого. 

3) Более того, я был отказался от выплаты GBG авторам, а сделал бы возможность пользователям эмитировать GBG в рамках 10% от их СГ. При этом надо продумать механизм, чтобы именно СГ была гарантией по GBG. А если курс голоса падает, то и % резерва СГ под гарантию GBG у пользователя растет.

Таким образом, мы получим реально стабильный коин. И пользователи будут понимать, что за эмиссию стабильного GBG они рискуют своей СГ.

А пока GBG - скам, когда декларируется цена стабильная в 1мг золота, а по факту нет, и эмиссия не остановлена при этом.

**Comment by afalaleev** [comment]
Тут путаница возникла - авторам перестает выплачиваться при достижении 5%.
Потом работает только эммиссия, которая должна создавать спрос, и это выбор инвестора сидеть с процентом в падающем в цене активе или выходить из него.

Вот с эммиссией явный косяк - односторонний инструмент без ограничений.

**Comment by t3ran13** [comment]
Ребята, тут вопрос гарантированного курса золотого, коьорый гарантирован в белой бумаге.

Все вопросы по эммисии можно обсуждать в таске #925 спецом для этого

**Comment by afalaleev** [comment]
Я согласен с @litrbooh, что основной актив сообщества - это не ГБГ, так как он пассивный актив, а как раз СГ, так как он определяет динамику сообщества. Отключение границы приведет к переливанию активов в пользу пассивных держателей на какой то срок, и я не вижу как это оживит платформу в перспективе.

Я понимаю, что хочет сделать @bitfag, но чтобы его схема заработала, надо саму сущность ГБГ переделать, сделав из ГБГ умный контракт, а не спрятав за ГБГ гарантии пассивных держателей залога.

**Comment by t3ran13** [comment]
@afalaleev 

>Я согласен с @litrbooh, что основной актив сообщества - это не ГБГ, так как он пассивный актив, а как раз СГ, так как он определяет динамику сообщества. Отключение границы приведет к переливанию активов в пользу пассивных держателей на какой то срок, и я не вижу как это оживит платформу в перспективе.

Напиши об этом #952, это не по теме гарнтии курса

>Я понимаю, что хочет сделать @bitfag, но чтобы его схема заработала, надо саму сущность ГБГ переделать, сделав из ГБГ умный контракт, а не спрятав за ГБГ гарантии пассивных держателей залога.

Если речь о стабилизации мин курса золотого через увеличение долга - это использование бага для временной решении проблемы. Будет работать пока курс голоса не станет еще меньше и мы не вернемся к той же ситуации. Какой смысл, это ведь не решение проблемы?

Решение очевидно - белая бумага это договор для любого инвестора в виде владельцев токенов голоса. И курс и логика его работы должна чоответсвовпть белой бумаге.

**Comment by bitfag** [comment]
I propose to replace min_price hard limit with a votable param and gradually adjust it, see https://goldvoice.club/@vvk/analyze-gbg-min-price-limit-removal/ for details (bottom)

**Comment by t3ran13** [comment]
Как по мне - медленно или сразу выход на стоимость обещанную в белой бумаге - не принципиально! важен сам факт, так что не вопрос, годнота!
